### PR TITLE
WIP 71 node tree builder

### PR DESCRIPTION
**Pull request by aburtsev**
## Related Issue(s)

Fixes #
#71

### CODE REVIEW

File: apps/web/pages/node/Node-react--type.i.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
+// temporary file. Until it'll be not merged in upstream master
```
Comment (webberwang) [inline apps/web/pages/node/Node-react--type.i.ts:1]: Already exists here https://github.com/codelab-ai/codelab.ai/blob/master/libs/shared/interface/node/src/subtypes/react/node-react--enum.ts

File: apps/web/pages/node/node.helpers.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { ReactNodeI } from '@codelab/shared/interface/node';
+
+interface ISelectOptionProps<T> {
+  value: T
+}
+
+export function generateSelectOption<T>(
```
Comment (webberwang) [inline apps/web/pages/node/node.helpers.ts:7]: Exists here https://github.com/codelab-ai/codelab.ai/blob/master/libs/components/ui/src/components/form/Form.types.ts

**Review by webberwang — CHANGES_REQUESTED** [review]

File: apps/web/pages/node/Node-react--type.i.ts (hunk 1 above)
Comment (aburtsev) [reply apps/web/pages/node/Node-react--type.i.ts:1]: I'll fix it

File: apps/web/pages/node/node.helpers.ts (hunk 2 above)
Comment (aburtsev) [reply apps/web/pages/node/node.helpers.ts:7]: I'll remove it and use the implementation from From.types.ts
Waiting for your decision about #119

Comment (webberwang) [reply apps/web/pages/node/node.helpers.ts:7]: Updated

**Comment by webberwang** [comment]
@aburtsev I refactored the organization a bit. A couple things I noticed.

- Breaking down a component into too many pieces is very difficult to follow. Utilize code folding & we can easily keep all the data required on the same file
- Use lower case naming for function props, as React throws an error
- For mapping data check out `tree-map.spec.tsx`

ButtonGroup_tsx_—_codelab__Workspace_

Hopefully this initial task gives you a better idea of how the project works. I've made some changes how types work will post update in #team on slack.

**Pull request merged**